Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,494 @@
+<?php
+
+/*
+ * This file is part of the package bk2k/bootstrap-package.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace BK2K\BootstrapPackage\DataProcessing;
+
+use BK2K\BootstrapPackage\Utility\LanguageUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Frontend\ContentObject\ContentDataProcessor;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+use TYPO3\CMS\Frontend\ContentObject\DataProcessorInterface;
+
+/**
+ * This menu processor utilizes HMENU to generate a json encoded menu
+ * string that will be decoded again and assigned to FLUIDTEMPLATE as
+ * variable. Additional DataProcessing is supported and will be applied
+ * to each record.
+ *
+ * Options:
+ * as - The variable to be used within the result
+ *
+ * See HMENU docs for more options.
+ * https://docs.typo3.org/typo3cms/TyposcriptReference/ContentObjects/Hmenu/Index.html
+ *
+ *
+ * Example TypoScript configuration:
+ *
+ * 10 = TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor
+ * 10 {
+ *   as = languagenavigation
+ *   dataProcessing {
+ *     10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+ *     10 {
+ *        references.fieldName = media
+ *     }
+ *   }
+ * }
+ */
+class LanguageMenuProcessor implements DataProcessorInterface
+{
+    const LINK_PLACEHOLDER = '###LINKPLACEHOLDER###';
+    const TARGET_PLACEHOLDER = '###TARGETPLACEHOLDER###';
+
+    /**
+     * The content object renderer
+     *
+     * @var ContentObjectRenderer
+     */
+    public $cObj;
+
+    /**
+     * The processor configuration
+     *
+     * @var array
+     */
+    protected $processorConfiguration;
+
+    /**
+     * Allowed configuration keys for menu generation, other keys
+     * will throw an exception to prevent configuration errors.
+     *
+     * @var array
+     */
+    public $allowedConfigurationKeys = [
```

Comment:
i think we can drop some more values here, since we need to ensure that this also works without `HMENU` at some point. having too much options that we would need to rebuild would make it for the core extremele hard to make changes here because its all public api.

```
cache_period
minItems
minItems.
maxItems
maxItems.
begin
begin.
alternativeSortingField
alternativeSortingField.
excludeUidList
excludeUidList.
excludeDoktypes
alwaysActivePIDlist
alwaysActivePIDlist.
protectLvar
addQueryString
addQueryString.
```

Reply:
agree, but i wasn't sure which one's i can drop, please make a suggestion

Reply:
well, the stuff above it the suggestion ;)

Reply:
oops

Reply:
done, but `alternativeSortingField` because it's supported now

Reply:
please close change request